src/include/distributed/multi_router_executor.h

```diff
@@ -9,16 +9,26 @@
 #ifndef MULTI_ROUTER_EXECUTOR_H_
 #define MULTI_ROUTER_EXECUTOR_H_
 
+#include "libpq-fe.h"
+
+#include "distributed/master_metadata_utility.h"
 #include "distributed/multi_physical_planner.h"
 #include "executor/execdesc.h"
 
 extern bool AllModificationsCommutative;
 
+typedef struct TxnParticipant
+{
+	ShardPlacement placement;
+	PGconn *connection;
+} TxnParticipant;
```

Comment:
Maybe I'm pedantic, but imo type definitions should be before variable declarations (after all, the latter might need the former).